src/assets/scripts/base/PositionModel.js

```diff
@@ -238,15 +204,13 @@ export default class PositionModel {
     }
 
     /**
-     * Fascade for `adjustForMagneticNorth`
-     *
-     * Adjusts x & y coordinates from true north to magnetic north
-     *
+     * Determine the `x` and `y` values of the `PositionModel`, used for drawing on the canvas
      * @for PositionModel
-     * @method adjustXYForMagneticNorth
+     * @method _calculateScreenPosition
+     * @private
      */
-    adjustXYForMagneticNorth() {
-        const { x, y } = adjustForMagneticNorth(this.x, this.y, this.magnetic_north);
+    _calculateScreenPosition() {
+        const [x, y] = PositionModel.calculatePosition(this.gps, this.reference_position, this.magnetic_north);
```

Comment:
theoretically, anything calling that static method _should_ already have a referencePosition.